app/lib/frontend/handlers.dart

```diff
@@ -109,7 +112,19 @@ shelf.Response docHandler(shelf.Request request) {
 
 /// Handles requests for /search
 Future<shelf.Response> searchHandler(shelf.Request request) async {
-  final queryText = request.url.queryParameters['q'];
+  String queryText = request.url.queryParameters['q'] ?? '';
+  String packagePrefix = request.url.queryParameters['pkg-prefix'];
+
+  final Match pkgMatch = _packageRegexp.firstMatch(queryText);
+  if (pkgMatch != null) {
+    // only if query attribute was not specified
+    if (packagePrefix == null) {
```

Comment:
So if the prefix is not explicitly given, add it – but only if the search string looks like a package name?